src/clientHandler.ts

```diff
@@ -12,298 +10,177 @@ import {
   ServerOptions,
   State,
 } from 'vscode-languageclient/node';
-import * as which from 'which';
-import { CUSTOM_BIN_PATH_OPTION_NAME, ServerPath } from './serverPath';
+import { ServerPath } from './serverPath';
 import { ShowReferencesFeature } from './showReferences';
-import { config, getFolderName, getWorkspaceFolder, normalizeFolderName, sortedWorkspaceFolders } from './vscodeUtils';
+import { config } from './vscodeUtils';
 
 export interface TerraformLanguageClient {
   commandPrefix: string;
   client: LanguageClient;
 }
 
-const MULTI_FOLDER_CLIENT = '';
-const clients: Map<string, TerraformLanguageClient> = new Map();
-
 /**
  * ClientHandler maintains lifecycles of language clients
- * based on the server's capabilities (whether multi-folder
- * workspaces are supported).
+ * based on the server's capabilities
  */
 export class ClientHandler {
   private shortUid: ShortUniqueId;
-  private supportsMultiFolders = true;
-
-  constructor(private lsPath: ServerPath, private reporter: TelemetryReporter) {
+  private tfClient: TerraformLanguageClient;
+  private commands: string[];
```

Comment:
```suggestion
  private commands: string[] = [];
```
and remove it in the constructor?